### DISCUSSION TITLE
Can plugins extend/modify the database?

### DISCUSSION THREAD

**Discussion by kiancross** [pretix/pretix#3478] [Q&A] [answered]
Is it possible for plugins to modify models? e.g., if I need a plugin to add a new field to a `Customer`.

I suppose another approach is to create a new table for the plugin, with a foreign key to the `Customer` and the new field stored in the new table?

What about signals such as the `item_formsets`? Does the `Item` model need extending for values in the formsets to be saved?

**Comment by raphaelm** [comment, marked as answer]
> Is it possible for plugins to modify models? e.g., if I need a plugin to add a new field to a Customer.

No, not possible.

> I suppose another approach is to create a new table for the plugin, with a foreign key to the Customer and the new field stored in the new table?

Yes.

> What about signals such as the item_formsets? Does the Item model need extending for values in the formsets to be saved?

No, it does not. There is no public usage example of item_formsets that I am aware of, but there is for item_forms, e.g. in pretix.plugins.badges
